pygfx/renderers/wgpu/_pipelinebuilder.py

```diff
@@ -28,12 +29,28 @@ def ensure_pipeline(renderer, wobject):
     new_pipeline_infos = None
 
     # Do we need to recreate the pipeline_objects?
-    if wobject.rev != wobject_pipeline.get("ref", ()):
+
+    # Check if render state has changed
+    state_hash = ""
+    for key in state:
+        values = state[key]
+        for value in values:
+            assert hasattr(value, "id")
```

Comment:
Is this assert really necessary? Does it ever happen?

Reply:
It's really unnecessary.  The value in the "render state dictionary" is put in by ourselves and controllable.